06_puppy/lexual/puppy.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+var out io.Writer = os.Stdout
+
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+// Storer defines standard CRUD operations for Puppys.
+type Storer interface {
+	CreatePuppy(*Puppy)
+	ReadPuppy(id uint32) *Puppy
+	UpdatePuppy(id uint32, puppy *Puppy)
+	DeletePuppy(id uint32) bool
+}
+
+// MapStore ////
+
+type MapStore map[uint32]Puppy
+
+// CreatePuppy stores a copy of Puppy p.
+func (store *MapStore) CreatePuppy(p *Puppy) {
+	(*store)[p.ID] = *p
+}
+
+// ReadPuppy retrieves a previously stored Puppy.
+func (store *MapStore) ReadPuppy(id uint32) *Puppy {
+	p, ok := (*store)[id]
+	if !ok {
+		return nil
+	}
+	return &p
```

Comment:
Little nit, and up to you, but if you flip the logic you can initialise on the if statement and save a line of code.

```go
if p, ok := (*store)[id]; ok {
  return &p
}
return nil
```